### PR TITLE
Require Ruby 2.3, add support for 2.7

### PR DESCRIPTION
**Pull request by BanzaiMan**
As this is a breaking change, it should be considered only for version 1.9.0.

Supersedes #543, #720.

### CODE REVIEW

**Comment by BanzaiMan** [comment]
See https://travis-ci.com/github/travis-ci/travis.rb/builds/160057418 for testing Faraday 1.0.

File: travis.gemspec [hunk 1]
```diff
@@ -368,17 +369,13 @@ Gem::Specification.new do |s|
   ]
 
   # dependencies
-  s.add_dependency "faraday",               "~> 0.9", "< 0.17.1" if RUBY_VERSION < "2.0"
-  s.add_dependency "faraday",               "~> 0.9" if RUBY_VERSION >= "2.0"
+  s.add_dependency "faraday",               "~> 0.9"
```
Comment (eregon) [inline travis.gemspec:372]: Could you update to 1.0 here? (#736)
Without that it's hard to use this gem efficiently as it basically enforces sequential blocking requests (https://github.com/socketry/async-http-faraday/issues/16)

Comment (BanzaiMan) [reply travis.gemspec:372]: As I stated in that ticket, it will be. :-)

**Comment by BanzaiMan** [comment]
We now have https://rubygems.org/gems/travis/versions/1.8.14.travis.1192.9 with the new requirements. If you would like, you can test this. I plan to have a new release (1.9.0) in the next few days.

**Comment by eregon** [comment]
I confirm that works :smiley:

**Comment by eregon** [comment]
Looking forward to the 1.9.0 release :)

**Comment by BanzaiMan** [comment]
I got sidetracked to an urgent issue. Planning a release next week.

**Pull request merged**